service/src/main/kotlin/fi/espoo/evaka/reports/ServiceNeedReport.kt

```diff
@@ -87,6 +100,60 @@ private fun Database.Read.getServiceNeedRows(date: LocalDate, authorizedUnits: A
         .toList()
 }
 
+private fun Database.Read.getServiceNeedRows2(date: LocalDate, authorizedUnits: AclAuthorization): List<ServiceNeedReportRow> {
+    // language=sql
+    val sql =
+        """
+        WITH ages AS (SELECT age FROM generate_series(0, 8) as age)
+        SELECT
+                care_area.name as care_area_name,
+                d.name as unit_name,
+                d.provider_type as unit_provider_type,
+                d.type as unit_type,
+                ages.age,
+        
+                count(DISTINCT p.id) FILTER ( WHERE sno.part_day = false ) as full_day,
+                count(DISTINCT p.id) FILTER ( WHERE sno.part_day = true ) as part_day,
+                count(DISTINCT p.id) FILTER ( WHERE sno.part_week = false ) as full_week,
+                count(DISTINCT p.id) FILTER ( WHERE sno.part_week = true ) as part_week,
+                count(DISTINCT p.id) FILTER ( WHERE sn.shift_care = true ) as shift_care,
+                count(DISTINCT p.id) FILTER ( WHERE sn is null ) as missing_service_need,
+                count(DISTINCT p.id) as total
+        FROM daycare d
+        JOIN ages ON true
+        JOIN care_area ON d.care_area_id = care_area.id
+        LEFT JOIN placement pl ON d.id = pl.unit_id AND daterange(pl.start_date, pl.end_date, '[]') @> :target_date AND pl.type != 'CLUB'::placement_type
+        LEFT JOIN person p ON pl.child_id = p.id AND date_part('year', age(p.date_of_birth)) = ages.age
+        LEFT JOIN new_service_need sn ON sn.placement_id = pl.id AND daterange(sn.start_date, sn.end_date, '[]') @> :target_date
+        LEFT JOIN service_need_option sno ON sno.id = sn.option_id
```

Comment:
this is copy paste apart from these lines 127-128 and `sn -> sno` on lines 115-118